### PR TITLE
Ports Goonchat V2

### PR DESCRIPTION
**Pull request by ThatLing**
Like #2703 but includes libvg made by PJB3005. Credits to @MoreRobustThanYou for the originally porting Goonchat. This makes it so non-ascii characters no longer make the message not show up. I haven't included the source for libvg but it can be found [here](https://github.com/vgstation-coders/vgstation13/tree/Bleeding-Edge/libvg).

Closes #2703.

:cl: MoreRobustThanYou and ThatLing
experimental: Ported goonchat from /tg/
/:cl:

### CODE REVIEW

**Comment by Kmc2000** [comment]
Speedmerge

**Comment by 1fbff5f83b23d39d38b1dfcb4cac8d9b** [comment]
Just stop

**Review by Amelia0010 — APPROVED** [review]
I'd say this is ready for a testmerge

**Comment by Alexandria404** [comment]
Does it have a toggle option whether you want the regular or goon chat?

**Comment by ThatLing** [comment]
@Alexandria404 Yes. Goonchat is enabled by default but can be disabled in preferences like the old PR.

File: code/modules/goonchat/browserOutput.dm [hunk 1]
```diff
@@ -0,0 +1,267 @@
...
+				if (!row || row.len < 3 || (!row["ckey"] || !row["compid"] || !row["ip"])) //Passed malformed history object
+					return
+				if (world.IsBanned(row["ckey"], row["compid"], row["ip"]))
+					found = row
+					break
+
+			//Uh oh this fucker has a history of playing on a banned account!!
+			if (found.len > 0)
+				//TODO: add a new evasion ban for the CURRENT client details, using the matched row details
+				message_admins("[key_name(src.owner)] has a cookie from a banned account! (Matched: [found["ckey"]], [found["ip"]], [found["compid"]])")
```
Comment (Amelia0010) [inline code/modules/goonchat/browserOutput.dm:190]: could you add a 
>webhook_send("cookies", "[key_name(src.owner)] has a cookie from a banned account! (Matched: [found["ckey"]], [found["ip"]], [found["compid"]])")
here please.

Comment (Amelia0010) [reply code/modules/goonchat/browserOutput.dm:190]: i'll add a yogbot endpoint to make it tell admuns

**Pull request closed without merging**